File: JAVA/src/swea/algorithmB/BinaryExpression.java
package JAVA.src.swea.algorithmB;


import java.util.Scanner;

public class BinaryExpression {

    public static void main(String[] args)  {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();

        for (int t = 1; t <= T; t++) {
            int N = sc.nextInt();
            int M = sc.nextInt();

            int bit = (1 << N) - 1;     // 마지막 N개의 비트를 모두 1로 표현하기
            String result = ((M & bit) == bit) ? "ON" : "OFF";

            System.out.println("#" + t + " " + result);
        }
    }
}

File: PYTHON/swea/AlgorithmB/BinaryExpression.py
import sys
sys.stdin = open("C:\\SSAFY\\algo-python\\PYTHON\\swea\\input.txt", "r")

T = int(input())

# 마지막 N개의 비트가 모두 1인지 체크

for t in range (T):
    N, M = map(int, input().split())
    result = ''
    
    bit = (1 << N) - 1
    result = 'ON' if (M & bit) == bit else 'OFF'
    
    print(f'#{t + 1} {result}')